src/System.Windows.Forms/src/misc/DpiHelper.cs

```diff
@@ -335,7 +335,166 @@ public static void ScaleButtonImageLogicalToDevice(Button button)
+                }
+            }
+
+            // For operating systems windows 8.1 to Windows 10 redstone 1 version.
+            else if (ApiHelper.IsApiAvailable(ExternDll.ShCore, nameof(SafeNativeMethods.GetProcessDpiAwareness)))
+            {
+                CAPS.PROCESS_DPI_AWARENESS processDpiAwareness;
+
+                SafeNativeMethods.GetProcessDpiAwareness(IntPtr.Zero, out processDpiAwareness);
+                switch (processDpiAwareness)
+                {
+                    case CAPS.PROCESS_DPI_AWARENESS.PROCESS_DPI_UNAWARE:
+                        return HighDpiMode.DpiUnaware;
+                    case CAPS.PROCESS_DPI_AWARENESS.PROCESS_SYSTEM_DPI_AWARE:
+                        return HighDpiMode.SystemAware;
+                    case CAPS.PROCESS_DPI_AWARENESS.PROCESS_PER_MONITOR_DPI_AWARE:
+                        return HighDpiMode.PerMonitor;
+                }
+            }
+
+            // For operating systems windows 7 to windows 8
+            else if (ApiHelper.IsApiAvailable(ExternDll.User32, nameof(SafeNativeMethods.IsProcessDPIAware)))
+            {
+                return SafeNativeMethods.IsProcessDPIAware() ?
+                       HighDpiMode.SystemAware :
+                       HighDpiMode.DpiUnaware;
+            }
+
+            // We should never get here, except someone ported this with force to < Windows 7.
+            return HighDpiMode.DpiUnaware;
+        }
+
+        /// <summary>
+        /// Sets the DPI awareness. If not available on the current OS, it falls back to the next possible.
+        /// </summary>
+        /// <returns>true/false - If the process DPI awareness is successfully set, returns true. Otherwise false.</returns>
+        internal static bool SetWinformsApplicationDpiAwareness(HighDpiMode highDpiMode)
+        {
+            NativeMethods.PROCESS_DPI_AWARENESS dpiFlag = NativeMethods.PROCESS_DPI_AWARENESS.PROCESS_DPI_UNINITIALIZED;
+
+            // For Windows 10 RS2 and above
+            if (ApiHelper.IsApiAvailable(ExternDll.User32, nameof(SafeNativeMethods.SetProcessDpiAwarenessContext)))
+            {
+                int rs2AndAboveDpiFlag;
+                switch (highDpiMode)
+                {
+                    case HighDpiMode.SystemAware:
+                        rs2AndAboveDpiFlag = NativeMethods.DPI_AWARENESS_CONTEXT_SYSTEM_AWARE;
+                        break;
+                    case HighDpiMode.PerMonitor:
+                        rs2AndAboveDpiFlag = NativeMethods.DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE;
+                        break;
+                    case HighDpiMode.PerMonitorV2:
+                        // Necessary for RS1, since this SetProcessDpiAwarenessContext IS available here.
+                        rs2AndAboveDpiFlag = NativeMethods.DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2;
+                        rs2AndAboveDpiFlag = SafeNativeMethods.IsValidDpiAwarenessContext(rs2AndAboveDpiFlag) ?
```

Comment:
```suggestion
                        rs2AndAboveDpiFlag = SafeNativeMethods.IsValidDpiAwarenessContext(NativeMethods.DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2) ?
```
I would not set this value twice like this

Reply:
True!